mm2src/coins/nft.rs

```diff
@@ -753,3 +804,59 @@ where
     }
     Ok(())
 }
+
+fn check_and_redact_if_spam(value: &mut Option<String>) -> bool {
+    match value {
+        Some(s) if s.contains("http://") || s.contains("https://") => {
```

Comment:
same here

Reply:
Why we need to check the order in this case? 
if token name or collection name etc contain http links or probably other links as shamardy [suggested](https://github.com/KomodoPlatform/atomicDEX-API/pull/1877#discussion_r1242866485) we mark it as spam.
it is not necessary that the link be complete in this case.

Reply:
You take string value and check it with contains. You are not checking the url scheme.

Reply:
> if token name or collection name etc contain http links or probably other links

So the value is not url, it just `might` include urls?

Reply:
> So the value is not url, it just `might` include urls?

yes, we check for urls those fields that should not contain links, as names or symbol. if contain then replace the text by `URL redacted for user protection`

Reply:
> Sorry I couldn't understand. Does this field `value: &mut Option<String>` contain url value or arbitrary text that could have urls in it? If it's always url, then we have to validate it properly(e.g. using `Url` abstraction type).

```rust
fn protect_from_history_spam(tx: &mut NftTransferHistory) {
    let collection_name_spam = check_and_redact_if_spam(&mut tx.collection_name);
    let token_name_spam = check_and_redact_if_spam(&mut tx.token_name);

    if collection_name_spam || token_name_spam {
        tx.possible_spam = true;
    }
}
```

collection_name and token_name should contain just text,  not any links, if `value` name of the param causes misunderstanding, lets rename it as `text`

Reply:
> if value name of the param causes misunderstanding, lets rename it as text

Yeah that could be great with simple doc-comment pointing "collection_name and token_name should contain just text, not any links"

Reply:
added doc comment for `check_and_redact_if_spam`. 
As for collection name, token name etc, I added doc comments above `protect_from_history_spam` and `protect_from_nft_spam` functions, as structures could have spam links if user dont set `protect_from_spam` as true.

Spam functionality is represented in check for spam functions, so lets keep doc comments for them.